### PR TITLE
[spirv][vulkan] Do not run nvidia gpu tests on buildkite

### PR DESCRIPTION
**Pull request by kuhar**
Issue: https://github.com/openxla/iree/issues/13123

### CODE REVIEW

File: build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ steps:
       - "buildkite-agent artifact download --step build build-artifacts.tgz ./"
       - "tar xzf build-artifacts.tgz"
       - "find build-android/ -name '*.cmake' -exec sed -i \"s!\\$IREE_DOCKER_WORKDIR/!\\$PWD/!g\" {} \\;"
-      - "ctest -j 4 --test-dir build-android/ --timeout 900 --output-on-failure --no-tests=error"
+      - 'ctest -j 4 --test-dir build-android/ --timeout 900 --output-on-failure --no-tests=error --label-exclude "^requires-gpu-nvidia$"'
```
Comment (ScottTodd) [inline build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml:28]: Another idea: we could have CMake skip defining tests with that label when `if(ANDROID)` is true (assuming there are no Android devices with NVIDIA GPUs... nvm? https://www.nvidia.com/en-us/shield/)

**Review by ScottTodd — APPROVED** [review]

**Review by pzread — APPROVED** [review]

**Comment by kuhar** [comment]
@ScottTodd @pzread Any idea what buildkite is complaining about?
> [2023-04-27T16:55:58Z] 2023-04-27 16:55:58 FATAL  Pipeline parsing of "pipeline.yml" failed (Expected identifier to start with a letter, got ")

**Comment by ScottTodd** [comment]
Two ideas:

1. flip the `"` and `'`
`- "so do 'this', with double quotes on the outside"`
`- 'and not "this", with double quotes on the inside'`

2. escape the `$`

**Comment by kuhar** [comment]
> Two ideas:
> 
> 1. flip the `"` and `'`
>    `- "so do 'this', with double quotes on the outside"`
>    `- 'and not "this", with double quotes on the inside'`
> 2. escape the `$`

Thanks! I opened https://github.com/openxla/iree/pull/13322. I couldn't find any clear documentation, but escaping `$` inside `'...'` doesn't seem to pass yml lint.

**Pull request merged**